[Bug] make - make <target> doesn't work

Version
```
$ kompile --version
RV-K version 1.0-SNAPSHOT
Git revision: 3f04459
Git branch: fix1646-ermsg
Build date: Mon Nov 30 17:49:34 UTC 2020
```

Description
-----------

`make <target>` doesn't work. <target> gets treated as a path instead of a target for Makefile

```
radumereuta@radu-mereuta:~/work/k/k-distribution/tests/regression-new$ time make -j 13 proofs
make -C proofs
make[1]: Entering directory '/home/radumereuta/work/k/k-distribution/tests/regression-new/proofs'
make -C deposit
make[2]: Entering directory '/home/radumereuta/work/k/k-distribution/tests/regression-new/proofs/deposit'
make -C test
make -C spec
make[3]: Entering directory '/home/radumereuta/work/k/k-distribution/tests/regression-new/proofs/deposit/test'
cat test.deposit.in 2>/dev/null | /home/radumereuta/work/k/k-distribution/bin/krun test.deposit --smt none -cTREEHEIGHT=3  -d . | diff - test.deposit.out
make[3]: Entering directory '/home/radumereuta/work/k/k-distribution/tests/regression-new/proofs/deposit/spec'
/home/radumereuta/work/k/k-distribution/bin/kprove deposit-spec.k --smt-prelude imap.smt2  -d . | diff - deposit-spec.k.out
make[3]: Leaving directory '/home/radumereuta/work/k/k-distribution/tests/regression-new/proofs/deposit/test'
[Warning] Compiler: Variable 'Size' defined but not used. Prefix variable name
with underscore if this is intentional.
        Source(/home/radumereuta/work/k/k-distribution/tests/regression-new/proofs/deposit/test/deposit.k)
        Location(103,28,103,32)
[Warning] Compiler: Variable 'M' defined but not used. Prefix variable name
with underscore if this is intentional.
        Source(/home/radumereuta/work/k/k-distribution/tests/regression-new/proofs/deposit/test/imap.k)
        Location(11,10,11,11)
[Warning] Compiler: Variable 'V' defined but not used. Prefix variable name
with underscore if this is intentional.
        Source(/home/radumereuta/work/k/k-distribution/tests/regression-new/proofs/deposit/test/imap.k)
        Location(12,20,12,21)
make[3]: Leaving directory '/home/radumereuta/work/k/k-distribution/tests/regression-new/proofs/deposit/spec'
make[2]: Leaving directory '/home/radumereuta/work/k/k-distribution/tests/regression-new/proofs/deposit'
make[1]: Leaving directory '/home/radumereuta/work/k/k-distribution/tests/regression-new/proofs'

real    0m7.272s
user    0m34.350s
sys     0m1.861s
radumereuta@radu-mereuta:~/work/k/k-distribution/tests/regression-new$
```

Expected Behavior
-----------------
Expecting this to recurse through every directory and run all tests that run `kprove`. 
I've deleted the `proofs` directory from and now I get a different error:
```
radumereuta@radu-mereuta:~/work/k/k-distribution/tests/regression-new$ make -j 13 proofs
make: *** No rule to make target 'proofs'.  Stop.
```

--- a/k-distribution/include/kframework/ktest-fail.mak
+++ b/k-distribution/include/kframework/ktest-fail.mak
@@ -13,7 +13,7 @@ KOMPILE_BACKEND?=llvm
 
 CHECK=| diff -
 
-.PHONY: kompile all clean update-results dummy
+.PHONY: kompile all clean update-results dummy krun proofs bmc searches strat kast
 
 # run all tests
 all: kompile

--- a/k-distribution/include/kframework/ktest-group.mak
+++ b/k-distribution/include/kframework/ktest-group.mak
@@ -1,13 +1,25 @@
 SUBCLEAN=$(addsuffix .clean,$(SUBDIRS))
 SUBUPDATE=$(addsuffix .update,$(SUBDIRS))
 SUBKOMPILE=$(addsuffix .kompile,$(SUBDIRS))
+SUBKRUN=$(addsuffix .krun,$(SUBDIRS))
+SUBPROOFS=$(addsuffix .proofs,$(SUBDIRS))
+SUBBMC=$(addsuffix .bmc,$(SUBDIRS))
+SUBSEARCHES=$(addsuffix .searches,$(SUBDIRS))
+SUBSTRAT=$(addsuffix .strat,$(SUBDIRS))
+SUBKAST=$(addsuffix .kast,$(SUBDIRS))
 
-.PHONY: all update-results clean $(SUBDIRS) $(SUBCLEAN) $(SUBUPDATE) $(SUBKOMPILE)
+.PHONY: all update-results clean kompile krun proofs bmc searches strat kast $(SUBDIRS) $(SUBCLEAN) $(SUBUPDATE) $(SUBKOMPILE) $(SUBKRUN) $(SUBPROOFS) $(SUBBMC) $(SUBSEARCHES) $(SUBSTRAT) $(SUBKAST)
 
 all: $(SUBDIRS)
 clean: $(SUBCLEAN)
 update-results: $(SUBUPDATE)
 kompile: $(SUBKOMPILE)
+krun: $(SUBKRUN)
+proofs: $(SUBPROOFS)
+bmc: $(SUBBMC)
+searches: $(SUBSEARCHES)
+strat: $(SUBSTRAT)
+kast: $(SUBKAST)
 
 $(SUBDIRS):
 	$(MAKE) -C $@
@@ -20,3 +32,21 @@ $(SUBUPDATE): %.update:
 
 $(SUBKOMPILE): %.kompile:
 	$(MAKE) -C $* kompile
+
+$(SUBKRUN): %.krun:
+	$(MAKE) -C $* krun
+
+$(SUBPROOFS): %.proofs:
+	$(MAKE) -C $* proofs
+
+$(SUBBMC): %.bmc:
+	$(MAKE) -C $* bmc
+
+$(SUBSEARCHES): %.searches:
+	$(MAKE) -C $* searches
+
+$(SUBSTRAT): %.strat:
+	$(MAKE) -C $* strat
+
+$(SUBKAST): %.kast:
+	$(MAKE) -C $* kast

--- a/k-distribution/include/kframework/ktest-keq.mak
+++ b/k-distribution/include/kframework/ktest-keq.mak
@@ -10,7 +10,7 @@ BASIC_SMT=$(abspath $(MAKEFILE_PATH)/z3/basic.smt2)
 
 CHECK=| diff -
 
-.PHONY: kompile keq all clean
+.PHONY: kompile keq all clean krun proofs bmc searches strat kast
 
 all: kompile keq
 

